public/app/core/services/backend_srv.ts

```diff
@@ -61,13 +96,65 @@ export class BackendSrv implements BackendService {
     });
   }
 
-  requestErrorHandler(err: any) {
+  private getFromFetchStream = (options: BackendSrvRequest) => {
+    const cleanParams = omitBy(options.params, v => v === undefined || v.length === 0);
+    const serializedParams = serializeParams(cleanParams);
+    const url = options.params ? `${options.url}?${serializedParams}` : options.url;
+    return fromFetch(url, {
+      method: options.method,
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json, text/plain, */*',
+        ...options.headers,
+      },
+      body: JSON.stringify(options.data),
+    }).pipe(
+      mergeMap(async response => {
+        const { status, statusText, ok } = response;
+        const textData = await response.text(); // this could be just a string
+        let data;
+        try {
+          data = JSON.parse(textData); // majority of the requests this will be something that can be parsed
+        } catch {
+          data = textData;
+        }
+        const fetchResponse: FetchResponse = { status, statusText, ok, data };
+        return fetchResponse;
+      }),
+      share() // sharing this so we can split into success and failure and then merge back
+    );
+  };
+
+  private toFailureStream = (options: BackendSrvRequest): MonoTypeOperatorFunction<FetchResponse> => $input =>
+    $input.pipe(
```

Comment:
Is the `$input` naming some kind of a convention?

Reply:
In the RxJs community they use `$` to  indicate that a variable is a stream. I've actually already removed it in the next commit :)